src/plugins/ssh-agent/index.ts

```diff
@@ -59,100 +56,71 @@ const asyncSpawn = async (
     }
   });
 
-/** Spawns a subprocess with the ssh-agent command.
- * Detects the auth socket and agent PID from stdout.
- * Stdout and stderr of the subprocess is printed to stderr in debug mode.
- * The returned promise resolves with an object that contains the auth socket and agent PID,
- * or rejects with the contents of stderr. */
-export const sshAgent = async (cmdArgs: AgentArgs) =>
-  new Promise<SshAgentEnv>((resolve, reject) => {
-    let stderr = "";
-    let stdout = "";
-
-    // There is a debug flag in ssh-agent but it causes the ssh-agent process to NOT fork
-    const child = spawn("ssh-agent");
-
-    child.stdout.on("data", (data) => {
-      const str = data.toString("utf-8");
-      if (cmdArgs.debug) {
-        print2(str);
-      }
-      stdout += str;
-    });
-
-    child.stderr.on("data", (data) => {
-      const str = data.toString("utf-8");
-      if (cmdArgs.debug) {
-        print2(str);
-      }
-      stderr += str;
-    });
-
-    const exitListener = child.on("exit", (code) => {
-      exitListener.unref();
-
-      if (code !== 0) {
-        return reject(stderr);
-      }
-
-      const authSockMatch = stdout.match(AUTH_SOCK_MESSAGE);
-      const agentPidMatch = stdout.match(AGENT_PID_MESSAGE);
+const isSshAgentRunning = async (args: AgentArgs) => {
+  try {
+    if (args.debug) print2("Searching for active ssh-agents");
+    await asyncSpawn(args, `pgrep`, ["-x", "ssh-agent"]);
```

Comment:
What if you have ssh-agent process(es) but your `SSH_AGENT_PID` is set to an invalid process ID that either doesn't exist or a different type of process. Would the actual ssh command try to use that?
Add a TODO if you think this is a valid edge case?

Reply:
TLDR: I'll write a TODO here. It's not an impossible edge case, but improbable.

Technically we don't use `SSH_AGENT_PID` for anything but we're just checking to make sure an ssh-agent is running. In theory its possible that:

1. someone starts an ssh-agent.
2. someone starts a new ssh-agent without changing the auth socket.
3. someone kills the old ssh agent.

but when they try to gain ssh access they'll get a permission denied error. if they run with a `--debug` flag it'll be very obvious that they're running the wrong agent/socket though. 

It wouldn't be impossible for us tho but it would take some `shell` magic to make sure that the PID + socket match and that they are both active.